cv_next/app/cv/[cvId]/components/commentSection/comment.tsx

```diff
@@ -1,78 +1,248 @@
+    </button>
+  ) : (
+    <button
+      className="text-green-500"
+      onClick={() => setResolvedCommentAction(true)}
+    >
+      <GoCheckCircle
+        style={{
+          transform: "translateY(2px)",
+        }}
+        fontSize="1.4rem"
+      />
+    </button>
+  );
+};
+
+const CommenterActions = ({
+  commenter,
   userId,
-  childDepth = 0,
-  cv,
-  commentsOfComment = [],
-  setCommentsOfComments,
-}: {
+  setShowAlert,
+  resolvedSection,
+}: any) => {
+  return commenter.id === userId ? (
+    <>
+      <span>
+        <>
+          <button className="text-red-500" onClick={() => setShowAlert(true)}>
+            <FaRegTrashCan
+              data-tooltip-target="Trash Icon"
+              style={{
+                transform: "translateY(2px)",
+              }}
+              fontSize="1.4rem"
+            />
+          </button>
+          <Tooltip id="Trash icon" message="delete"></Tooltip>
+          <span> </span>
+        </>
+      </span>
+      <span> </span>
+      {resolvedSection}
+    </>
+  ) : null;
+};
+
+const GeneralActions = ({ startNewComment, votingSection, comment }: any) => {
+  return (
+    <>
+      <span>{startNewComment}</span>
+      <span>
+        {votingSection} {comment.upvotes?.length || null}
+      </span>
+    </>
+  );
+};
+
+const AlertComponent = ({ showAlert, onAlertClick }: any) => {
+  return showAlert ? (
+    <Alert
+      display={showAlert ? "flex" : "none"}
+      message="You sure you want to delete this comment?"
+      color="red"
+      onClick={onAlertClick}
+    ></Alert>
+  ) : null;
+};
+
+const StartNewComment = ({
+  commentOnCommentStatus,
+  setCommentOnCommentStatus,
+}: any) => {
+  return !commentOnCommentStatus ? (
+    <button className="text-green-500">
+      <FaComment
+        size="1.2rem"
+        style={{
+          marginRight: "0.4rem",
+          transform: "translateY(2px)",
+        }}
+        onClick={() => setCommentOnCommentStatus(!commentOnCommentStatus)}
+      />
+    </button>
+  ) : null;
+};
+
+interface CommentProps {
   comment: CommentModel;
   userId: string;
   childDepth?: number;
-  cv: CvModel;
-  commentsOfComment: any[];
+  commentsOfComment: CommentModel[];
   setCommentsOfComments: (
     update: (prev: Map<string, any[]>) => Map<string, any[]>
   ) => void;
-}) {
+}
+
+export default function Comment({
+  comment,
+  userId,
+  childDepth = 0,
+  commentsOfComment = [],
+  setCommentsOfComments,
+}: CommentProps) {
   const [commentOnCommentStatus, setCommentOnCommentStatus] =
     useState<boolean>(false);
   const [commentOnComment, setCommentOnComment] = useState<string>("");
-  const router = useRouter();
   const [showAlert, setShowAlert] = useState<boolean>(false);
   const onAlertClick = async (type: boolean) => {
     if (type) await deleteCommentAction();
     setShowAlert(false);
   };
-  const supabase = useSupabase();
+  const { mutate } = useSWRConfig();
 
   const addNewCommentClickEvent = async () => {
```

Comment:
Please remember that this _entire function_ might get recreated on every render cycle (i.e. view interaction) if not memoized; please use `useMemo` or `useCallback` (whichever you think is more appropriate).

Reply:
useCallback is for not recreating function so I went with that